### PR TITLE
[Ernie GPU Optim]: Fuse three fc to multihtead matmul

### PR DESCRIPTION
**Pull request by NHZlX**
Fuse three Fc to mutlithead matmul op and combined three Fc to one Fc.
Under cuda10.1 p4 card, the normal ernie model performance is improved from 10ms to 9.5ms.

Original:

image

Now:

image

### CODE REVIEW

File: paddle/fluid/operators/fused/multihead_matmul_op.cu [hunk 1]
```diff
@@ -312,96 +312,167 @@ void MatMulWithHeadQKV(const platform::CUDADeviceContext &context, int head_num,
...
+  const int S = gridDim.x;
+  const int B = gridDim.y;
+  const int NH = N * H;
+  const int NHS = NH * S;
+  const int in_offset = n * H + m * NH + s * 3 * NH + b * NHS * 3;
+  const int bias_offset = m * NH + n * H;
+  const int out_offset = s * H + n * S * H + b * NHS + m * NHS * B;
+
+  const int i = threadIdx.x;
+  if (i < H) {
```
Comment (zhaoyuchen2018) [inline paddle/fluid/operators/fused/multihead_matmul_op.cu]: 这个if是不是可以去掉？

File: paddle/fluid/operators/fused/multihead_matmul_op.cu [hunk 2]
```diff
@@ -312,96 +312,167 @@ void MatMulWithHeadQKV(const platform::CUDADeviceContext &context, int head_num,
...
+}
 
-  int m = batch_size * seq_len;
-  int k = head_num * size_per_head;
+void TransQKVWithBias(const int batch, const int seq_len, const int head_size,
+                      const int head_num, const float *input, const float *bias,
+                      float *output, cudaStream_t stream) {
+  // BxSx3xNxH + 3xNxH -> 3xBxNxSxH
+  const dim3 grid(seq_len, batch, 3);
+  if (head_size % 2 == 0) {
```
Comment (zhaoyuchen2018) [inline paddle/fluid/operators/fused/multihead_matmul_op.cu]: 这里为啥没有用float4呢，128byte应该也能放在cacheline吧

Comment (NHZlX) [reply paddle/fluid/operators/fused/multihead_matmul_op.cu]: done!

File: paddle/fluid/operators/fused/multihead_matmul_op.cu (hunk 1 above)
Comment (NHZlX) [reply paddle/fluid/operators/fused/multihead_matmul_op.cu]: done!

**Review by zhaoyuchen2018 — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM to PADDLE_ENFROCE*

**Pull request merged**